qiskit/transpiler/passes/layout/dense_layout.py

```diff
@@ -42,10 +45,29 @@ def __init__(self, coupling_map, backend_prop=None):
         super().__init__()
         self.coupling_map = coupling_map
         self.backend_prop = backend_prop
-        self.cx_mat = None
-        self.meas_arr = None
-        self.num_cx = 0
-        self.num_meas = 0
+        num_qubits = 0
+        self.adjacency_matrix = None
+        if self.coupling_map:
+            num_qubits = self.coupling_map.size()
+            self.adjacency_matrix = retworkx.adjacency_matrix(self.coupling_map.graph)
+        self.error_mat = np.zeros((num_qubits, num_qubits))
+        if self.backend_prop and self.coupling_map:
+            error_dict = {
+                tuple(gate.qubits): gate.parameters[0].value
+                for gate in self.backend_prop.gates
+                if len(gate.qubits) == 2
+            }
+            for edge in self.coupling_map.get_edges():
+                gate_error = error_dict.get(edge)
+                if gate_error is not None:
+                    self.error_mat[edge[0]][edge[1]] = gate_error
+            for index, qubit_data in enumerate(self.backend_prop.qubits):
+                # Handle faulty qubits edge case
```

Comment:
Can you clarify what this loop is doing?

Reply:
Ah, the comment here is for the if statement below specifically (when faulty qubits are set, the `qubits in properties` != set of qubits` in coupling map). So it's just bailing if we have more qubits to read from the noise properties than are in the coupling map.

This loop is to get the measurement error for each qubit in the coupling map, in `BackendV1` (and `BaseBackend`) this is contained in the `qubits` field of `BackendProperties` object under the dict where `obj["name"] == "readout_error"`. It's a bit more intuitive in `BackendV2` with the `Target`, but I was saving that for a follow on PR.